src/node_report.cc

```diff
@@ -487,48 +490,231 @@ void GetNodeReport(Isolate* isolate, DumpEvent event, const char* message, const
+      // TODO timeout is not actually public however it is present in
+      // all current versions of libuv. Once uv_timer_get_timeout is
+      // in a supported level of libuv we should test for it with dlsym
+      // and use it instead, in case timeout moves in the future.
+      data << ", timeout in: "
+#ifdef _WIN32
+           << (handle->timer.due - uv_now(handle->timer.loop)) << " ms";
+#else
+           << (handle->timer.timeout - uv_now(handle->timer.loop)) << " ms";
+#endif
+      break;
+    }
+    case UV_TTY: {
+      int height, width, rc;
+      type = "tty";
+      rc = uv_tty_get_winsize(&(handle->tty), &width, &height);
+      if (rc == 0) {
+        data << "width: " << width << ", height: " << height;
+      }
+      break;
+    }
+    case UV_UDP: {
+      type = "udp";
+      reportEndpoints(h, data);
+      break;
+    }
+    case UV_SIGNAL: {
+      // SIGWINCH is used by libuv so always appears.
+      // See http://docs.libuv.org/en/v1.x/signal.html
+      type = "signal";
+      data << "signum: " << handle->signal.signum
+      // node::signo_string() is not exported by Node.js on Windows.
+#ifndef _WIN32
+           << " (" << node::signo_string(handle->signal.signum) << ")"
+#endif
+           ;
+      break;
+    }
     case UV_FILE: type = "file"; break;
     // We shouldn't see "max" type
-    case UV_HANDLE_TYPE_MAX : break;
+    case UV_HANDLE_TYPE_MAX : type = "max"; break;
   }
 
-  snprintf(buf, sizeof(buf),
-#ifdef _WIN32
-              "[%c%c]   %-10s0x%p\n",
-#else
-              "[%c%c]   %-10s%p\n",
+  if (h->type == UV_TCP || h->type == UV_UDP
+#ifndef _WIN32
+      || h->type == UV_NAMED_PIPE
+#endif
+      ) {
+    // These *must* be 0 or libuv will set the buffer sizes to the non-zero
+    // values they contain.
+    int send_size = 0;
+    int recv_size = 0;
+    if (h->type == UV_TCP || h->type == UV_UDP) {
+      data << ", ";
+    }
+    uv_send_buffer_size(h, &send_size);
+    uv_recv_buffer_size(h, &recv_size);
+    data << "send buffer size: " << send_size
+         << ", recv buffer size: " << recv_size;
+  }
+
+  if (h->type == UV_TCP || h->type == UV_NAMED_PIPE || h->type == UV_TTY ||
+      h->type == UV_UDP || h->type == UV_POLL) {
+    uv_os_fd_t fd_v;
+    uv_os_fd_t* fd = &fd_v;
+    int rc  = uv_fileno(h, fd);
+    // uv_os_fd_t is an int on Unix and HANDLE on Windows.
+#ifndef _WIN32
+    if (rc == 0) {
+      switch (fd_v) {
+      case 0:
+        data << ", stdin"; break;
+      case 1:
+        data << ", stdout"; break;
+      case 2:
+        data << ", stderr"; break;
+      default:
+        data << ", file descriptor: " << static_cast<int>(fd_v);
+        break;
+      }
+    }
 #endif
-              uv_has_ref(h)?'R':'-',
-              uv_is_active(h)?'A':'-',
-              type.c_str(), (void*)h);
+  }
+
+  if (h->type == UV_TCP || h->type == UV_NAMED_PIPE || h->type == UV_TTY) {
+
+    data << ", write queue size "
```

Comment:
Minor: could you add the usual colon after 'write queue size' to separate the name from the value, for consistency